src/main/java/com/training/web/pages/flipkart/PaymentPage.java

```diff
@@ -44,4 +49,38 @@ public String getEmiSupport(String bankName) {
     public String getTitle() {
         return driver.getTitle();
     }
+
+    public PaymentPage scrollToPageEnd() {
+        $x(pageEndElement).shouldBe(Condition.visible).scrollIntoView(true);
+        return this;
+    }
+
+    public PaymentPage clickBackToTop() {
+        $x(backToTopIcon).shouldBe(Condition.visible).click();
+        return this;
+    }
+
+    public boolean getBackToTopIconVisibility(){
+        $x(backToTopIcon).shouldBe(Condition.disappear);
+        return $x(backToTopIcon).isDisplayed();
+    }
+
+    public boolean getHeaderVisibility() {
+        SelenideElement element= $x(header);
+        WebDriver driver = element.getWrappedDriver();
+
+        return (Boolean)((JavascriptExecutor)driver).executeScript(
+                "var elem = arguments[0],                 " +
+                        "  box = elem.getBoundingClientRect(),    " +
+                        "  cx = box.left + box.width / 2,         " +
+                        "  cy = box.top + box.height / 2,         " +
+                        "  e = document.elementFromPoint(cx, cy); " +
+                        "for (; e; e = e.parentElement) {         " +
+                        "  if (e === elem)                        " +
+                        "    return true;                         " +
+                        "}                                        " +
+                        "return false;                            "
+                , element);
```

Comment:
I am thinking of having this code in BaseScreen Class with method : `isElementInVisibleInView(Selenide element)`
So basically `getHeaderVisibility()` should make a call to this method in BaseScreen class